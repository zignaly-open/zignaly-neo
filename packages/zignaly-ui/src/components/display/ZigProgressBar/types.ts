export type ZigProgressBarProps = {
  value: number;
  className?: string;
  max?: number;
};
