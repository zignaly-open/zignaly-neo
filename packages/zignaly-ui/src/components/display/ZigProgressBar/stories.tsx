import * as React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import ZigProgressBar from ".";
import styled from "styled-components";

const Container = styled.div`
  width: 300px;
`;

export default {
  title: "Display/ZigProgressBar",
  component: ZigProgressBar,
} as ComponentMeta<typeof ZigProgressBar>;

const Template: ComponentStory<typeof ZigProgressBar> = (args) => (
  <Container>
    <ZigProgressBar {...args} />
  </Container>
);

const defaultProps = {
  value: 25,
};

export const ZigProgressBarBar = Template.bind({});
ZigProgressBarBar.args = {
  ...defaultProps,
  max: 50,
};
