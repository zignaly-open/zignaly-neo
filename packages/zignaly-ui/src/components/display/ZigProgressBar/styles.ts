import croma from "chroma-js";
import styled from "styled-components";
import { styled as muiStyled } from "@mui/system";
import ZigTypography from "../ZigTypography";

export const Layout = styled.div`
  position: relative;
  padding-top: 10px;
  margin-bottom: 12px;
  display: flex;
  align-items: center;
  width: 100%;
`;

export const Bar = styled.div`
  background: linear-gradient(90deg, #567734 0%, #948d23 34.43%, #8e3b24 68.81%, #8c1954 99.89%);
  box-shadow: inset 0px 1px 1px -1px rgba(73, 9, 123, 0.25);
  border-radius: 100px;
  height: 4px;
  width: 100%;
  display: flex;
  align-items: center;
  position: relative;
  margin: 0 7px;
`;

export const DotContainer = styled.div<{ value: number }>`
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  position: absolute;
  left: ${(props) => props.value}%;
  top: -4px;
  transform: translate(-50%, 0%);
`;

export const Dot = styled.div`
  background: rgba(255, 255, 255, 0.9);
  height: 7px;
  margin-top: 3px;
  width: 2px;
`;

export const Label = muiStyled(ZigTypography)`
  color: RGB(193, 193, 200);

  &.body1 {
    font-size: 13px;
  }
`;

const gradient = croma.scale(["#567734", "#948d23", "#8e3b24", "#8c1954"]);

export const LabelTooltip = styled(Label)<{ value: number }>`
  border-radius: 3px;
  pointer-events: none;
  background: ${(props) =>
    gradient(props.value / 100)
      .brighten(1)
      .toString()};

  top: -38px;
  line-height: 1 !important;
  padding: 5px 6px;
  position: relative;
  display: block;
  color: ${(props) =>
    gradient(props.value / 100)
      .darken(1.5)
      .toString()};

  &:before {
    content: "";
    border: 4px solid transparent;
    border-top-color: ${(props) =>
      gradient(props.value / 100)
        .brighten(1)
        .toString()};
    position: absolute;
    top: 100%;
    left: calc(50% - 4px);
  }

  small {
    font-size: 10px;
    bottom: 2px;
    position: relative;
  }
`;
