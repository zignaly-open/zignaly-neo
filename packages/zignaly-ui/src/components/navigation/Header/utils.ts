// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
