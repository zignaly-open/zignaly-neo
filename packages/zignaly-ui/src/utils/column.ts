import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
