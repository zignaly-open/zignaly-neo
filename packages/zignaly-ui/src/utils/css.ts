import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
