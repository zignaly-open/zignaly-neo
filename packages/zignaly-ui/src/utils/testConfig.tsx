import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
