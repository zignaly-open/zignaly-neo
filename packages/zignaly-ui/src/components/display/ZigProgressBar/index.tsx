import * as React from "react";
import { ReactElement } from "react";

import { Bar, Dot, DotContainer, Label, LabelTooltip, Layout } from "./styles";

import { ZigProgressBarProps } from "./types";

// TODO: rename to ZigZigProgressBar
function ZigProgressBar({ className, value, max = 100 }: ZigProgressBarProps): ReactElement {
  const normalizedValue = (100 * Math.min(value, max)) / max;
  return (
    <Layout className={className}>
      <Label>0%</Label>
      <Bar>
        <DotContainer value={normalizedValue}>
          <Dot />
          <LabelTooltip value={normalizedValue}>
            {value}
            <small>%</small>
          </LabelTooltip>
        </DotContainer>
      </Bar>
      <Label>{max}%</Label>
    </Layout>
  );
}

export default ZigProgressBar;
