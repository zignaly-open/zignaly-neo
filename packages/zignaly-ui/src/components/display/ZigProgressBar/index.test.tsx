import React from "react";
import ZigProgressBar from "./index";
import { renderWithProvidersUi } from "../../../utils/testConfig";

describe("components/display/ZigProgressBar", () => {
  it("renders", async () => {
    const { container } = renderWithProvidersUi(
      <ZigProgressBar className={"test-class"} value={10} />,
    );

    const progressBar = container.querySelector(".test-class") as Element;
    expect(progressBar).toBeVisible();
    const progressBarStyles = getComputedStyle(progressBar);
    expect(progressBar).toMatchSnapshot();
    expect(progressBarStyles).toMatchSnapshot();
  });
});
