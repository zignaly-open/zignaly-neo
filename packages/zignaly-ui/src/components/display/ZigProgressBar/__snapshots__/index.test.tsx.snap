// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/ZigProgressBar renders 1`] = `
<div
  class="sc-bczRLJ ipRuFC test-class"
>
  <span
    class="MuiTypography-root MuiTypography-body1 css-l00qwn-MuiTypography-root"
  >
    0%
  </span>
  <div
    class="sc-gsnTZi bbeEYO"
  >
    <div
      class="sc-dkzDqf jYByLK"
      value="10"
    >
      <div
        class="sc-hKMtZM jhmszV"
      />
      <span
        class="MuiTypography-root MuiTypography-body1 sc-eCYdqJ cVgrQQ css-l00qwn-MuiTypography-root"
        value="10"
      >
        10
        <small>
          %
        </small>
      </span>
    </div>
  </div>
  <span
    class="MuiTypography-root MuiTypography-body1 css-l00qwn-MuiTypography-root"
  >
    100
    %
  </span>
</div>
`;

exports[`components/display/ZigProgressBar renders 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "position",
  "2": "padding-top",
  "3": "margin-bottom",
  "4": "align-items",
  "5": "width",
  "6": "visibility",
  "_importants": {
    "align-items": "",
    "display": "",
    "margin-bottom": "",
    "padding-top": "",
    "position": "",
    "visibility": undefined,
    "width": "",
  },
  "_length": 7,
  "_onChange": [Function],
  "_values": {
    "align-items": "center",
    "display": "flex",
    "margin-bottom": "12px",
    "padding-top": "10px",
    "position": "relative",
    "visibility": "visible",
    "width": "100%",
  },
}
`;
